tests/kernel/interrupt/CMakeLists.txt

```diff
@@ -17,3 +17,8 @@ target_sources(app PRIVATE
 
 target_sources_ifdef(CONFIG_DYNAMIC_INTERRUPTS app PRIVATE src/dynamic_isr.c)
 target_sources_ifdef(CONFIG_X86 app PRIVATE src/regular_isr.c)
+target_sources_ifdef(CONFIG_SHARED_INTERRUPTS app PRIVATE src/static_shared_irq.c)
+
+if (CONFIG_SHARED_INTERRUPTS)
+target_sources_ifdef(CONFIG_DYNAMIC_INTERRUPTS app PRIVATE src/dynamic_shared_irq.c)
```

Comment:
Doesn't seem right for the static test which can be run w/o `CONFIG_DYNAMIC_INTERRUPTS=y`. Also, this would require some extra `#ifdef` logic in the dynamic test's code. I guess the "proper" way of handling this would be to split the shared IRQ test into its static and dynamic parts, each of them having a testcase in `testcase.yaml`. Apart from it being messier, it would increase the number of twister tests.